.github/workflows/lint-ruby.yml

```
@@ -5,37 +5,37 @@ on:
       - 'dependabot/**'
     paths:
       - 'Gemfile*'
-      - '.rubocop.yml'
+      - '.rubocop*.yml'
       - '**/*.rb'
       - '**/*.rake'
       - '.github/workflows/lint-ruby.yml'
 
   pull_request:
     paths:
       - 'Gemfile*'
-      - '.rubocop.yml'
+      - '.rubocop*.yml'
       - '**/*.rb'
       - '**/*.rake'
       - '.github/workflows/lint-ruby.yml'
 
 jobs:
   lint:
     runs-on: ubuntu-latest
+
     steps:
-      - name: Checkout Code
+      - name: Clone repository
         uses: actions/checkout@v3
+
+      - name: Install native Ruby dependencies
+        run: sudo apt-get install -y libicu-dev libidn11-dev
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
         with:
-          fetch-depth: 0
+          bundler-cache: true
```

Use pined ruby version
```suggestion
          ruby-version: .ruby-version
          bundler-cache: true
```